6.x/crud-filters.md

```diff
@@ -5,63 +5,176 @@
+
+```php
+// whenActive method, applied immediately
+CRUD::filter('active')
+	->type('simple')
+	->whenActive(function ($value) {
+	    $this->crud->addClause('where', 'active', '1');
+	})->apply();
+
+// whenActive, left to be applied by the operation itself
+CRUD::filter('active')
+	->type('simple')
+	->whenActive(function ($value) {
+	    $this->crud->addClause('where', 'active', '1');
+	});
+```
+
+- ```->whenInactive(function($value) {})``` - By chaining **whenInactive()** on a filter you define what should be done when that filter is NOT active; it accepts a closure that is called when the filter is NOT active - either immediately by further chaining ```apply()``` on the filter, or automatically after all filters are defined, by the List operation itself; you can also use ```->else()```, ```->fallbackLogic()```, ```->whenNotActive()```, ```->ifInactive()``` and ```->ifNotActive()``` which are its aliases:
+
+```php
+// fallbackLogic method, applied immediately
+CRUD::filter('active')
+	->type('simple')
+	->whenActive(function ($value) {
+	    $this->crud->addClause('where', 'active', '1');
+	})->whenInactive(function ($value) {
+	    $this->crud->addClause('where', 'active', '0');
+	})->apply();
+
+// else alias, left to be applied by the operation itself
+CRUD::filter('active')
+	->type('simple')
+	->label('Simple')
+	->whenActive(function ($value) {
+	    $this->crud->addClause('where', 'active', '1');
+	})->else(function ($value) {
+	    $this->crud->addClause('where', 'active', '0');
+	});
+```
+
+- ```->apply()``` - By chaining **apply()** on a filter after you've specified the filtering logic using `whenActive()` or `whenInactive()`, you immediately call the appropriate closure; in most cases this isn't necessary, because the List operation automatically performs an `apply()` on all filters; but in some cases, where the filtering closures want to stop or change execution, you can use `apply()` to have that logic applied before the next bits of code get executed;
+
+<a name="fluent-filters-other-chained-methods"></a>
+#### Other Chained Methods
+
+If you chain the following methods to a ```CRUD::filter('name')```, they will do something very specific instead of adding that attribute to the filter:
+
+- ```->remove()``` - By chaining **remove()** on a filter you remove it from the current operation;
+
+```php
+CRUD::filter('name')->remove();
+```
+
+- ```->forget('attribute_name')``` - By chaining **forget('attribute_name')** to a filter you remove that attribute from the filter;
+
+```php
+CRUD::filter('name')->forget('suffix');
```

Comment:
@tabacitu 
- `suffix` attribute is not used in any filter. Applicable example make more sense.
- forget not working for the `before` attribute. & When I tried to the `forget('label')` This made the label empty, where it is supposed to pick from filter `name`
- No idea how it works with `ifActive` OR `whenActive` (two-word-attribute)
- `CRUD::filter('name)->label('Full Name');` is missing in Other Chained Methods section

Reply:
```suggestion
CRUD::filter('price')->prefix('$'); // will have "$" as prefix 
CRUD::filter('price')->forget('prefix'); // will no longer have "$" as prefix
```

Reply:
> forget not working for the before attribute. & When I tried to the forget('label') This made the label empty, where it is supposed to pick from filter name
> No idea how it works with ifActive OR whenActive (two-word-attribute)

That makes sense. Forget will only forget attributes, it won't work on methods or aliases. 
- `before()` isn't an attribute, it's a method;
- `whenActive()` isn't an attribute, it's a method, an alias to `logic()`
- `whenInactive()` isn't an attribute, it's a method, an alias to `fallbackLogic()`

So:
- before cannot be forgotten
- logic/fallbacklogic can be forgotten with their attributes `logic` / `fallbackLogic`

You think we should document this here, as a note?

Reply:
I updated my suggestion above to cover this, let me know what you think.

Reply:
@tabacitu Everything is perfect here but no filter has a prefix attribute